src/prototype_rust_sdk/workflow_context.rs

```diff
@@ -335,38 +390,76 @@ impl ChildWorkflowOptions {
 #[derive(Default, Debug)]
 pub struct ChildWorkflow {
     opts: ChildWorkflowOptions,
-    sequence_number: Option<u32>,
+}
+
+pub struct PendingChildWorkflow {
+    pub(crate) child_wf_cmd_seq_num: u32,
+    pub status: ChildWorkflowStartStatus,
+}
+
+impl PendingChildWorkflow {
+    /// Returns `None` if the child did not start successfully. The returned [StartedChildWorkflow]
+    /// can be used to wait on, signal, or cancel the child workflow.
+    pub fn as_started(&self) -> Option<StartedChildWorkflow> {
+        match &self.status {
+            ChildWorkflowStartStatus::Succeeded(s) => Some(StartedChildWorkflow {
+                child_wf_cmd_seq_num: self.child_wf_cmd_seq_num,
+                run_id: s.run_id.clone(),
+            }),
+            _ => None,
+        }
+    }
+}
+
+pub struct StartedChildWorkflow {
+    pub(crate) child_wf_cmd_seq_num: u32,
+    pub run_id: String,
 }
 
 impl ChildWorkflow {
     /// Start the child workflow, the returned Future is cancellable.
-    pub fn start(
-        &mut self,
-        cx: &mut WfContext,
-    ) -> impl CancellableFuture<ChildWorkflowStartStatus> {
-        let seq = cx.next_child_workflow_sequence_number;
+    pub fn start(&mut self, cx: &mut WfContext) -> impl CancellableFuture<PendingChildWorkflow> {
+        let child_seq = cx.next_child_workflow_sequence_number;
         cx.next_child_workflow_sequence_number += 1;
-        self.sequence_number = Some(seq);
-        let (cmd, unblocker) = WFCommandFut::new(CancellableID::ChildWorkflow(seq));
+        let (cmd, unblocker) = WFCommandFut::new(CancellableID::ChildWorkflow {
+            child_seq,
+            cancel_cmd_seq: None,
+        });
         cx.send(
             CommandCreateRequest {
-                cmd: self.opts.to_command(seq).into(),
+                cmd: self.opts.to_command(child_seq).into(),
                 unblocker,
             }
             .into(),
         );
         cmd
     }
+}
 
-    /// Wait on completion of a child workflow execution started with `start_child_workflow`.
-    /// The returned Future is cancellable.
+impl StartedChildWorkflow {
+    /// Create a future that will wait until completion of this child workflow execution
```

Comment:
We might want to add an `execute` method too that combines `start` and `result` like we have in the other SDKs.